indico/modules/events/forms.py

```diff
@@ -135,3 +136,42 @@ class UnlistedEventsForm(IndicoForm):
     authorized_creators = PrincipalListField(_('Authorized users'), [HiddenUnless('enabled', preserve_data=True)],
                                              allow_external_users=True, allow_groups=True,
                                              description=_('These users may create unlisted events.'))
+
+
+class DataRetentionSettingsForm(IndicoForm):
+    minimum_data_retention = TimeDeltaField(_('Minimum data retention period'), units=('weeks',),
+                                            description=_('Specify the minimum data retention period (in weeks) '
+                                                          'configurable for registrations. This includes the '
+                                                          'retention period of individual registration form fields.'),
+                                            render_kw={'placeholder': _('Indefinite'), 'min': 1, 'max': 521})
+    maximum_data_retention = TimeDeltaField(_('Maximum data retention period'), units=('weeks',),
+                                            description=_(
+        'Specify the maximum data retention period (in weeks) configurable for registrations. This includes the '
+        'retention period of individual registration form fields. Note that setting this value will make the '
+        'retention period field mandatory during registration form setup.'),
+        render_kw={'placeholder': _('Indefinite'), 'min': 1, 'max': 521})
+
+    def validate_minimum_data_retention(self, field):
+        maximum_retention = self.data.get('maximum_data_retention')
+        if maximum_retention and not (minimum_retention := self._validate_data_rentention(field)):
+            raise ValidationError(_('The minimum retention period cannot be indefinite when a maximum is set.'))
+        if maximum_retention and minimum_retention > maximum_retention:
+            raise ValidationError(_('The minimum retention period cannot be greater than the maximum.'))
+
+    def validate_maximum_data_retention(self, field):
+        if not (maximum_retention := self._validate_data_rentention(field)):
+            return
+        minimum_retention = self.data.get('minimum_data_retention')
+        if minimum_retention and minimum_retention > maximum_retention:
+            raise ValidationError(_('The minimum retention period cannot be greater than the maximum.'))
+
+    def _validate_data_rentention(self, field):
+        retention_period = field.data
+        if retention_period is None:
+            return
+        if retention_period <= timedelta():
+            raise ValidationError(_('The retention period cannot be zero or negative.'))
+        elif retention_period > timedelta(days=3650):
```

Comment:
Same here, this is an admin setting, no need to restrict this

Reply:
Ah I see, we had 10 years on the event-level field before. I think there it made more sense.

I was about to suggest to set the default maximum to 10 years, but then you cannot choose Indefinite for an event anymore, so not a good idea...

Reply:
Right, I wanted to keep the 10 years to be consistent with how it was in the event, but since it's now configurable I guess it doesn't need to be restricted here

Reply:
I'm thinking that if we remove the maximum of 10 years, and an instance does not configure the min or max data retention, (so it can be indefinite) then the event mangers will be allowed to choose a retention period of any length for their registration forms, unless they leave it empty for indefinite.
So basically, removing the max of 10 years from the configuration would also remove the max from the registration forms? Or the 10 year max will be removed from the settings but still enforced on the registration form? If it's the later then it might as well remain in the settings since the only places using the values from the settings are the registration form and fields.

Reply:
TLDR; Should I remove the maximum restriction of 10 years from the settings?

Reply:
Yeah I think we can get rid of it.